### PR TITLE
For CASSANDRA-10593, test that archived commit logs can be restored m…

### PR DESCRIPTION
**Pull request by aweisberg**
…ore than once without corruption

### CODE REVIEW

File: snapshot_test.py [hunk 1]
```diff
@@ -323,3 +327,63 @@ def run_archive_commitlog(self, restore_point_in_time=False, restore_archived_co
             shutil.rmtree(system_col_snapshot_dir)
             debug("removing tmp_commitlog: " + tmp_commitlog)
             shutil.rmtree(tmp_commitlog)
+
+    def test_archive_and_restore_commitlog_repeatedly(self):
+        """Run archive commit log restoration test repeatedly to make sure it is idempoten
+           and doesn't fail if done repeatedly"""
```
Comment (ptnapoleon) [inline snapshot_test.py]: nit: `idempotent`

Also, the `"""` should be on their own lines. Thanks!

File: snapshot_test.py [hunk 2]
```diff
@@ -323,3 +327,63 @@ def run_archive_commitlog(self, restore_point_in_time=False, restore_archived_co
...
+
+        cluster = self.cluster
+        cluster.populate(1)
+        (node1,) = cluster.nodelist()
```
Comment (ptnapoleon) [inline snapshot_test.py]: `node1 = cluster.nodelist()[0]` is better looking. And even if you disagree, the `()` aren't necessary here.

File: snapshot_test.py [hunk 3]
```diff
@@ -323,3 +327,63 @@ def run_archive_commitlog(self, restore_point_in_time=False, restore_archived_co
...
+
+        # Create a temp directory for storing commitlog archives:
+        tmp_commitlog = safe_mkdtemp()
+        debug("tmp_commitlog: " + tmp_commitlog)
```
Comment (ptnapoleon) [inline snapshot_test.py]: Use String interpolation please. `debug("tmp_commitlog: {}".format(tmp_commitlog))`

File: snapshot_test.py [hunk 4]
```diff
@@ -323,3 +327,63 @@ def run_archive_commitlog(self, restore_point_in_time=False, restore_archived_co
...
+
+        try:
+            # Check that there are at least one commit log backed up that
+            # is not one of the active commit logs:
+            commitlog_dir = os.path.join(node1.get_path(), 'commitlogs')
+            debug("node1 commitlog dir: " + commitlog_dir)
+
+            cluster.flush()
+
+            self.assertTrue(len(set(os.listdir(tmp_commitlog)) - set(os.listdir(commitlog_dir))) > 0)
```
Comment (ptnapoleon) [inline snapshot_test.py]: `self.assertGreater(len(.....), 0)` would provide better debug output

File: snapshot_test.py [hunk 5]
```diff
@@ -323,3 +327,63 @@ def run_archive_commitlog(self, restore_point_in_time=False, restore_archived_co
...
+            node1.stop()
+            node1.start(wait_for_binary_proto=True)
+
+            # Shouldn't be any additional data since it's replaying the same stuff repeatedly
+            session = self.patient_cql_connection(node1)
+
+            rows = session.execute('SELECT count(*) from ks.cf')
+            self.assertEqual(rows[0][0], 30000)
+        finally:
+            debug("removing tmp_commitlog: " + tmp_commitlog)
```
Comment (ptnapoleon) [inline snapshot_test.py]: String interpolation here again, please

File: snapshot_test.py [hunk 6]
```diff
@@ -323,3 +327,63 @@ def run_archive_commitlog(self, restore_point_in_time=False, restore_archived_co
             shutil.rmtree(system_col_snapshot_dir)
             debug("removing tmp_commitlog: " + tmp_commitlog)
             shutil.rmtree(tmp_commitlog)
+
+    def test_archive_and_restore_commitlog_repeatedly(self):
+        """Run archive commit log restoration test repeatedly to make sure it is idempoten
```
Comment (ptnapoleon) [inline snapshot_test.py]: Can you put, on its own line, `@jira_ticket CASSANDRA-10593` in this docstring?

**Comment by ptnapoleon** [comment]
Test logic lgtm. Just a few style complaints.

**Comment by mambocab** [comment]
Ping @aweisberg, can you make these style changes?

**Comment by ptnapoleon** [comment]
I'm going to merge and make the style changes

**Pull request merged**